tss-esapi/src/abstraction/transient.rs

```diff
@@ -494,6 +421,117 @@ impl TransientKeyContext {
         }
         Ok(())
     }
+
+    /// Given the parameters for an asymmetric key, return its [Public] structure
+    ///
+    /// The public part of the key can optionally be inserted in the structure.
+    ///
+    /// # Errors
+    /// * if the public key and the parameters don't match, `InconsistentParams` is returned
+    fn get_public_from_params(params: KeyParams, pub_key: Option<PublicKey>) -> Result<Public> {
+        let decrypt_flag = matches!(
+            params,
+            KeyParams::Rsa {
+                scheme: RsaScheme::RsaEs,
+                ..
+            } | KeyParams::Rsa {
+                scheme: RsaScheme::Oaep(..),
+                ..
+            }
+        );
+        let object_attributes = ObjectAttributesBuilder::new()
+            .with_fixed_tpm(true)
+            .with_fixed_parent(true)
+            .with_sensitive_data_origin(true)
+            .with_user_with_auth(true)
+            .with_decrypt(decrypt_flag)
+            .with_sign_encrypt(true)
+            .with_restricted(false)
+            .build()?;
+
+        let mut pub_builder = PublicBuilder::new()
+            .with_public_algorithm(match params {
+                KeyParams::Ecc { .. } => PublicAlgorithm::Ecc,
+                KeyParams::Rsa { .. } => PublicAlgorithm::Rsa,
+            })
+            .with_name_hashing_algorithm(HashingAlgorithm::Sha256)
+            .with_object_attributes(object_attributes);
+        match params {
+            KeyParams::Rsa {
+                size,
+                scheme,
+                pub_exponent,
+            } => {
+                let unique = match pub_key {
+                    None => PublicKeyRsa::default(),
```

Comment:
Feel free to ignore this but that I had some difficulties reading that match-expression. So I came up with an alternative way of writing that might work. But if something is hard or5 easy to read is a completely subjective matter.

```rust
let unique = possible_pub_key
    .map(|pub_key| {
        if let PublicKey::Rsa(val) = pub_key {
            PublicKeyRsa::try_from(val)
        } else {
            Err(Error::local_error(ErrorKind::InconsistentParams))
        }
    })
    .transpose()?
    .unwrap_or_default();
```

Reply:
Done